### ISSUE TITLE
Reducing the readahead / output buffer size in encoding mode?

### ISSUE THREAD

**Issue by twilson90** [mpv-player/mpv#8908]
If I run mpv in encoding mode (--o=...) and pipe the output to ffplay or ffmpeg with the -re flag (realtime flag) it reads ahead (depending on the machine I'm running it on) between 5 to10 seconds.
So if I monitor the time-pos, on playing a media file it quickly shoots up to 5-10 seconds (meaning the output buffer is full presumably) and then time-pos increments in more-or-less realtime, but the output video (if I'm outputting to ffplay or rtmp) will always be 10 seconds behind what time-pos says.
If I play a video file with a still background (or not very much movement), time-pos can shoot up to 60 seconds almost immediately, and then I have a 60 second delay between my input action and the result from the encoding, suggesting the output buffer has a fixed length.

For my purposes I want to make this delay as low as possible.

It's been a while since I tried, but in the past I tried reducing this length of time with various command line arguments in mpv and ffmpeg/ffplay and got no where.
And I never got to the bottom of whether the issue was with ffmpeg or mpv, so apologies if this is obviously an ffmpeg issue.

Any ideas?

**Comment by TheAMM** [comment]
Add the `realtime` filter to mpv to limit its speed. What you're seeing is the encoder buffer, muxer buffer, pipe buffer and ffmpeg's decode buffer - scraping that latency down will be quite an adventure for you. You may try to give mpv and ffmpeg the nobuffer flag (among other things), though I forget which lavf/lavd/etc option it was on mpv.

**Comment by hedgehog90** [comment]
I forgot to mention I tried the realtime and arealtime filters, and although it cut the latency down to a solid 3 seconds (on any media), it also led to some fairly edge-case weirdness that I never got to the bottom of.
If there was any pause in the encoding (any scenario that would lead to buffering) it would mess up the entire stream from there onwards, audio sync issues, pausing, etc.
Suspect I'm running into an issue to do with mpv internal.

I've tried `-fflag=+nobuffer`  for ffmpeg and `--demuxer-lavf-o-add=fflags=+nobuffer` for mpv but see no noticeable difference unfortunately.
